frontend/src/store/gym/types.ts

```diff
@@ -0,0 +1,78 @@
+export type GymStatusStates = {
+  gymIsOpen: boolean | null
+  avatar: string
+  keyHolder: { displayName: string; telegramHandle: string }
+  keyIsReturned: boolean
+}
+
+export type ButtonStates = {
+  keyWithMe: boolean
+  returnKey: boolean
+  toggleGym: boolean
+}
+
+// Added string type to handle some edge cases (specifically toggleGym).
+export type ButtonTypes = string | 'keyWithMe' | 'returnKey' | 'closeGym' | 'openGym'
+
+export type HistoryEntry = {
+  date: string
+  details: UserEntry[]
+}
+
+export type UserEntry = {
+  statusChange: string
+  requesttime: number
+  userDetails: string
+}
+
+export type KeyWithArgs = {
+  userID: string
```

Comment:
standardise ID to Id

Reply:
this one a little harder to change same as `requesttime` because we didn't really catch this during development and it's in backend alr, might be a little troublesome to change